File: apps/pos/native-ios/DesktopKitchenPOS/Theme/AppColors.swift
import SwiftUI

enum AppColors {
    // Backgrounds
    static let background = Color(hex: 0x0A0A0A)    // neutral-950
    static let card = Color(hex: 0x171717)            // neutral-900
    static let cardHover = Color(hex: 0x1C1C1C)
    static let surface = Color(hex: 0x262626)         // neutral-800

    // Borders
    static let border = Color(hex: 0x262626)          // neutral-800
    static let borderLight = Color(hex: 0x404040)     // neutral-700

    // Accent — dynamically overridable via branding
    nonisolated(unsafe) static var accent = Color(hex: 0xDC2626)          // red-600
    nonisolated(unsafe) static var accentDark = Color(hex: 0xB91C1C)      // red-700
    nonisolated(unsafe) static var accentLight = Color(hex: 0xEF4444)     // red-500

    // Text
    static let textPrimary = Color.white
    static let textSecondary = Color(hex: 0xA3A3A3)   // neutral-400
    static let textTertiary = Color(hex: 0x737373)    // neutral-500
    static let textMuted = Color(hex: 0x525252)       // neutral-600

    // Status
    static let success = Color(hex: 0x16A34A)         // green-600
    static let successLight = Color(hex: 0x4ADE80)    // green-400
    static let warning = Color(hex: 0xF59E0B)         // amber-500
    static let warningLight = Color(hex: 0xFBBF24)    // amber-400
    static let error = Color(hex: 0xDC2626)           // red-600
    static let info = Color(hex: 0x2563EB)            // blue-600

    // Role badges
    static let roleAdmin = Color(hex: 0xDC2626)       // red
    static let roleManager = Color(hex: 0x9333EA)     // purple
    static let roleKitchen = Color(hex: 0x2563EB)     // blue
    static let roleCashier = Color(hex: 0x16A34A)     // green

    // MARK: - Dynamic Branding

    /// Replace the accent color from a hex string (e.g. "#0d9488").
    static func applyBranding(hex: String) {
        guard let rgb = parseHexString(hex) else { return }
        accent = Color(.sRGB, red: rgb.r, green: rgb.g, blue: rgb.b, opacity: 1)
        // Derive darker / lighter shades
        accentDark = Color(.sRGB, red: rgb.r * 0.8, green: rgb.g * 0.8, blue: rgb.b * 0.8, opacity: 1)
        accentLight = Color(.sRGB, red: min(rgb.r * 1.2, 1), green: min(rgb.g * 1.2, 1), blue: min(rgb.b * 1.2, 1), opacity: 1)
    }

    private static func parseHexString(_ hex: String) -> (r: Double, g: Double, b: Double)? {
        var cleaned = hex.trimmingCharacters(in: .whitespacesAndNewlines)
        if cleaned.hasPrefix("#") { cleaned.removeFirst() }
        guard cleaned.count == 6, let value = UInt(cleaned, radix: 16) else { return nil }
        return (
            r: Double((value >> 16) & 0xFF) / 255,
            g: Double((value >> 8) & 0xFF) / 255,
            b: Double(value & 0xFF) / 255
        )
    }
}

extension Color {
    init(hex: UInt, opacity: Double = 1.0) {
        self.init(
            .sRGB,
            red: Double((hex >> 16) & 0xFF) / 255,
            green: Double((hex >> 8) & 0xFF) / 255,
            blue: Double(hex & 0xFF) / 255,
            opacity: opacity
        )
    }
}

File: apps/pos/native-windows/DesktopKitchenPOS/Theme/AppColors.cs
using Microsoft.UI;
using Microsoft.UI.Xaml.Media;
using Windows.UI;

namespace DesktopKitchenPOS.Theme;

public static class AppColors
{
    // Backgrounds
    public static readonly Color Background = Color.FromArgb(255, 10, 10, 10);     // neutral-950
    public static readonly Color Card = Color.FromArgb(255, 23, 23, 23);            // neutral-900
    public static readonly Color CardHover = Color.FromArgb(255, 28, 28, 28);
    public static readonly Color Surface = Color.FromArgb(255, 38, 38, 38);         // neutral-800

    // Borders
    public static readonly Color Border = Color.FromArgb(255, 38, 38, 38);          // neutral-800
    public static readonly Color BorderLight = Color.FromArgb(255, 64, 64, 64);     // neutral-700

    // Accent — dynamically overridable via branding
    public static Color Accent = Color.FromArgb(255, 220, 38, 38);                  // red-600
    public static Color AccentDark = Color.FromArgb(255, 185, 28, 28);              // red-700
    public static Color AccentLight = Color.FromArgb(255, 239, 68, 68);             // red-500

    // Text
    public static readonly Color TextPrimary = Colors.White;
    public static readonly Color TextSecondary = Color.FromArgb(255, 163, 163, 163); // neutral-400
    public static readonly Color TextTertiary = Color.FromArgb(255, 115, 115, 115);  // neutral-500
    public static readonly Color TextMuted = Color.FromArgb(255, 82, 82, 82);        // neutral-600

    // Status
    public static readonly Color Success = Color.FromArgb(255, 22, 163, 74);         // green-600
    public static readonly Color SuccessLight = Color.FromArgb(255, 74, 222, 128);   // green-400
    public static readonly Color Warning = Color.FromArgb(255, 245, 158, 11);        // amber-500
    public static readonly Color WarningLight = Color.FromArgb(255, 251, 191, 36);   // amber-400
    public static readonly Color Error = Color.FromArgb(255, 220, 38, 38);           // red-600
    public static readonly Color Info = Color.FromArgb(255, 37, 99, 235);            // blue-600

    // Brushes (for XAML binding convenience)
    public static SolidColorBrush BackgroundBrush => new(Background);
    public static SolidColorBrush CardBrush => new(Card);
    public static SolidColorBrush SurfaceBrush => new(Surface);
    public static SolidColorBrush BorderBrush => new(Border);
    public static SolidColorBrush AccentBrush => new(Accent);
    public static SolidColorBrush TextPrimaryBrush => new(TextPrimary);
    public static SolidColorBrush TextSecondaryBrush => new(TextSecondary);
    public static SolidColorBrush TextTertiaryBrush => new(TextTertiary);
    public static SolidColorBrush SuccessBrush => new(Success);
    public static SolidColorBrush WarningBrush => new(Warning);
    public static SolidColorBrush ErrorBrush => new(Error);
    public static SolidColorBrush InfoBrush => new(Info);

    public static void ApplyBranding(string hex)
    {
        var color = ParseHex(hex);
        if (color == null) return;

        Accent = color.Value;
        AccentDark = Color.FromArgb(255,
            (byte)(color.Value.R * 0.8),
            (byte)(color.Value.G * 0.8),
            (byte)(color.Value.B * 0.8));
        AccentLight = Color.FromArgb(255,
            (byte)Math.Min(color.Value.R * 1.2, 255),
            (byte)Math.Min(color.Value.G * 1.2, 255),
            (byte)Math.Min(color.Value.B * 1.2, 255));
    }

    private static Color? ParseHex(string hex)
    {
        var cleaned = hex.TrimStart('#');
        if (cleaned.Length != 6) return null;
        if (!uint.TryParse(cleaned, System.Globalization.NumberStyles.HexNumber, null, out var value))
            return null;
        return Color.FromArgb(255,
            (byte)((value >> 16) & 0xFF),
            (byte)((value >> 8) & 0xFF),
            (byte)(value & 0xFF));
    }
}
